homeassistant/core.py

```diff
@@ -1409,7 +1409,10 @@ def __init__(
 
         self.entity_id = entity_id
         self.state = state
-        self.attributes = ReadOnlyDict(attributes or {})
+        if type(attributes) is not ReadOnlyDict:  # noqa: E721
```

Comment:
Add a comment why we use `type`and not `isinstance`.

Reply:
13f3249849bc2fd15a991a2e866fadfd61871321